frontend/src/store/modules/page_timings/actions.js

```diff
@@ -0,0 +1,122 @@
+    FETCH_TABLE_DATA,
+    FETCH_PAGE_DATA
+} from "./types/actions";
+import {
+    SET_SELECTED_PERIOD,
+    SET_ACTIVE_BUTTON,
+    SET_GROUPED_PARAMETER,
+    RESET_BUTTON_FETCHING,
+    SET_BUTTON_FETCHING,
+    SET_BUTTON_DATA,
+    RESET_TABLE_FETCHING,
+    SET_TABLE_FETCHING,
+    RESET_LINE_CHART_FETCHING,
+    SET_LINE_CHART_FETCHING,
+    SET_LINE_CHART_DATA,
+    SET_TABLE_DATA,
+} from "./types/mutations";
+
+import {factoryPageTimingsService} from '@/api/page_timings/factoryPageTimingsService';
+import {getTimeByPeriod} from '@/services/periodService';
+
+export default {
+    [CHANGE_SELECTED_PERIOD]: (context, payload) => {
+        context.commit(SET_SELECTED_PERIOD, payload.value);
+        context.dispatch(FETCH_PAGE_DATA);
+    },
+    [CHANGE_ACTIVE_BUTTON]: (context, button) => {
+        context.commit(SET_ACTIVE_BUTTON, button);
+        context.dispatch(FETCH_LINE_CHART_DATA);
+        context.dispatch(FETCH_TABLE_DATA);
+
+    },
+    [CHANGE_FETCHED_BUTTON_STATE]: (context, data) => {
+        if (data.value) {
+            context.commit(SET_BUTTON_FETCHING, data.button);
+        } else {
+            context.commit(RESET_BUTTON_FETCHING, data.button);
+        }
+    },
+    [FETCH_BUTTON_DATA]: (context, type) => {
+        context.commit(SET_BUTTON_FETCHING, type);
+
+        const period = getTimeByPeriod(context.state.selectedPeriod);
+        const startDate = period.startDate;
+        const endDate = period.endDate;
+
+        return factoryPageTimingsService.create(type)
+            .fetchButtonValue(startDate.unix(), endDate.unix())
+            .then(response => {
+                context.commit(SET_BUTTON_DATA, {button: type, value: response.value});
+                context.commit(RESET_BUTTON_FETCHING, type);
+            })
+            .catch(err => {
+                context.commit(RESET_BUTTON_FETCHING, type);
+                throw err;
+            });
+    },
+    [CHANGE_FETCHED_LINE_CHART_STATE]: (context, value) => {
+
+        if (value) {
+            context.commit(SET_LINE_CHART_FETCHING);
+        } else {
+            context.commit(RESET_LINE_CHART_FETCHING);
+        }
+    },
+    [FETCH_LINE_CHART_DATA]: (context) => {
+        context.commit(SET_LINE_CHART_FETCHING);
+        const period = getTimeByPeriod(context.state.selectedPeriod);
+        const startDate = period.startDate;
+        const endDate = period.endDate;
+
+        return factoryPageTimingsService.create(context.state.activeButton)
+            .fetchChartValues(startDate.unix(), endDate.unix(), period.interval)
+            .then(data => context.commit(SET_LINE_CHART_DATA, data))
+            .finally(() => context.commit(RESET_LINE_CHART_FETCHING));
+
+    },
+    [CHANGE_GROUPED_PARAMETER]: (context, parameter) => {
+        context.commit(SET_GROUPED_PARAMETER, parameter);
+        context.dispatch(FETCH_TABLE_DATA);
+    },
+    [CHANGE_FETCHED_TABLE_STATE]: (context, value) => {
```

Comment:
and probably this one too

Reply:
in task for linking particular component can delete unused methods